ProcessMaker/processmaker
# FOUR-8392 Template Automation test

## Issue & Reproduction Steps
To perform the Template Automation test, we need to ensure that all [process-templates](https://github.com/ProcessMaker/process-templates) are correctly imported into PM4. Additionally, we must confirm that these templates work properly when creating a Process.

## Solution
- Created a unit test that will fetch and count all the processes within [process-templates](https://github.com/ProcessMaker/process-templates) and compare them with the ones on the Default Templates database.
- This test will also check all templates to make sure they can be added as Processes.

## How to Test
To successfully configure the automated default template synchronization, follow these steps:

- Open your `.env` file for your project.
- Add the following new variables and their respective values:
  - The DEFAULT_TEMPLATE_BRANCH will be the reference of the UnitTest
```
DEFAULT_TEMPLATE_REPO=process-templates
DEFAULT_TEMPLATE_BRANCH=task/FOUR-8392
DEFAULT_TEMPLATE_CATEGORIES=all
```
- install the new templates with the artisan command OR Wipe and create a new PM installation
-- Install: `php artisan processmaker:sync-default-templates`
-- Wipe: `php artisan db:wipe --force && php artisan migrate:fresh --seed --force`
- Make sure all the packages are installed.
- Run PHPUnit `tests/Feature/Templates/Api/ProcessTemplateTest.php
- When the new installation is created, check if the new processes are in the templates tab.
- Create a Process from the new templates.

## Related Tickets & Packages
- Ticket [FOUR-8392](https://processmaker.atlassian.net/browse/FOUR-8392)
- [New templates ](https://github.com/ProcessMaker/process-templates/pull/5)

ci:processmaker:task/FOUR-8392

## Code Review Checklist


[FOUR-8392]: https://processmaker.atlassian.net/browse/FOUR-8392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
